src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_SIZE = 10;
+    private int size = 0;
+    private int index;
+    private Object[] array = new Object[START_ARRAY_SIZE];
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            array = Arrays.copyOf(array, (int) (array.length * 1.5));
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index: "
+                    + index + ", Size: " + size);
+        } else {
+            if (size == array.length) {
+                array = Arrays.copyOf(array, (int) (array.length * 1.5));
+            }
+            Object[] secondPart = Arrays.copyOfRange(array, index, size);
+            array[index] = value;
+            System.arraycopy(secondPart, 0, array, index + 1, secondPart.length);
+            size++;
```

Comment:
instead of all of this, you just need one System.arraycopy call, then lines 31,33